substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SecuritySubstitutions.java

```diff
@@ -135,7 +142,10 @@
     @Substitute
     private static <T> T doPrivilegedWithCombiner(PrivilegedExceptionAction<T> action, AccessControlContext context, Permission... perms) throws Throwable {
         try {
-            return action.run();
+            AccessControllerUtil.accStack.get().push(context);
+            T result = action.run();
```

Comment:
What happens when this throws an exception, we could make an infinite stack? I think we should us try with resources to make sure this is closed.